core/src/main/java/org/elasticsearch/index/fielddata/ScriptDocValues.java

```diff
@@ -340,7 +340,7 @@ public void setNextDocId(int docId) throws IOException {
                 resize(in.docValueCount());
                 for (int i = 0; i < count; i++) {
                     GeoPoint point = in.nextValue();
-                    values[i].reset(point.lat(), point.lon());
+                    values[i] = new GeoPoint(point.lat(), point.lon());
```

Comment:
I'm wondering whether we should keep things this way here and do the cloning in `get(int index)`/`getValue()` to help GC by having even shorter lived objects, and potentially make escape analysis more likely to not ever create those objects.